apps/site/lib/site_web/controllers/schedule/line/helpers.ex

```diff
@@ -80,10 +80,36 @@ defmodule SiteWeb.ScheduleController.Line.Helpers do
   defp do_get_branch_route_stops(route, direction_id, route_pattern_id) do
     route.id
     |> get_route_patterns(direction_id, route_pattern_id)
-    |> Enum.filter(&by_typicality(&1, route_pattern_id))
+    |> filter_by_min_typicality()
+    # The below row is for testing purposes for the ferry ONLY
+    # This is needed until a specific routepattern has a distinct typicality
+    |> temporary_hh_ferry_only_func(route.id)
     |> Enum.map(&stops_for_route_pattern/1)
   end
 
+  @spec filter_by_min_typicality([RoutePattern.t()]) :: [RoutePattern.t()]
+  defp filter_by_min_typicality(route_patterns) do
+    route_patterns
+    |> Enum.reduce({nil, []}, fn
+      %RoutePattern{typicality: typicality} = r, {t, _a} when typicality < t -> {typicality, [r]}
+      %RoutePattern{typicality: typicality} = r, {typicality, a} -> {typicality, [r | a]}
+      _, acc -> acc
+    end)
+    |> elem(1)
+    |> Enum.reverse()
+  end
+
+  @spec filter_by_min_typicality([RoutePattern.t()]) :: [RoutePattern.t()]
+  defp temporary_hh_ferry_only_func(route_patterns, id) do
+    if id == "Boat-F1" do
```

Comment:
Rather than an `if` statement, you can pattern match in the function head:
```elixir
defp temporary_hh_ferry_only_func([route_pattern | _], "Boat-F1") do
  [route_pattern]
end
defp temporary_hh_ferry_only_func(route_patterns, _route_id) do
  route_patterns
end
```

Reply:
Oh so clean!  I restructured `do_get_branch_route_stops` to contain the if-else, so the matching option went away, I think.   But great to know for next time